src/clj/fluree/db/query/api.cljc

```diff
@@ -207,36 +207,36 @@
         (dataset db-map defaults)))))
 
 (defn query-connection-fql
-  [conn query]
+  [conn query {:keys [role did]}]
   (go-try
-    (let [{query :subject, did :did} (or (<? (cred/verify query))
-                                         {:subject query})
-          {:keys [t opts] :as query*}  (update query :opts sanitize-query-options did)
+    (let [{query :subject, did :did}  (or (<? (cred/verify credentialed-opts-query))
+                                          {:subject query})
+          {:keys [t opts] :as query*} (update query :opts sanitize-query-options did)
 
           default-aliases (some-> query* :from util/sequential)
           named-aliases   (some-> query* :from-named util/sequential)]
       (if (or (seq default-aliases)
               (seq named-aliases))
-        (let [s-ctx       (ctx-util/extract query)
-              ds          (<? (load-dataset conn default-aliases named-aliases t
+        (let [s-ctx    (ctx-util/extract query)
+              ds       (<? (load-dataset conn default-aliases named-aliases t
                                             s-ctx opts))
-              query**     (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
-              max-fuel    (:max-fuel opts)]
+              query**  (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
+              max-fuel (:max-fuel opts)]
           (if (::util/track-fuel? opts)
             (<? (track-query ds max-fuel query**))
             (<? (fql/query ds query**))))
         (throw (ex-info "Missing ledger specification in connection query"
                         {:status 400, :error :db/invalid-query}))))))
 
-
 (defn query-connection-sparql
-  [conn query]
+  [conn query {:keys [role did] :as request-opts}]
   (go-try
-    (let [fql (sparql/->fql query)]
-      (<? (query-connection-fql conn fql)))))
+    (let [fql              (sparql/->fql query)
```

Comment:
You'll need to do a `verify` before this call, as you may not have an actual sparql string for a query right here.

Something like this:
```
(let [{sparql :subject, did :did}  (or (<? (cred/verify credentialed-opts-query))
                                                          {:subject query})
         fql  (sparql/->fql query)]
    ....)
```

Reply:
That feels confusing, because this function will only get called if the request's `Content-Type` is set to `application/sparql-query`. So even for sparql queries, the query itself can be submitted as a JWS string?

Reply:
All the way down here we don't know that this function call is the result of an http request - that's all handled by `server`. So while we know that `server` only calls this function after verifying the query, there will be other callers that are not `server`. So we need to call `cred/verify` in this path as well.

Reply:
Ok, I think this must have been broken previously then, because we weren't verifying any credentials here previously. This will also make it so that we get the query in `query-connection`, and if it's sparql, then we verify the credentials in the sparql->fql step and then verify the credentials again in `query-connection-fql`. But maybe I'm just not understanding what exactly our verification process for credentials is doing or when it's necessary.

Reply:
Yeah, we actually verify multiple times. If the thing we're verifying isn't a credential or a jws, then `cred/verify` just returns `nil`, so that's why it's always wrapped in an `or` expression. And I think you're right that it was broken before, I don't think we were really considering authentication for sparql queries.